app/src/main/java/com/twilio/video/app/VideoService.kt

```diff
@@ -4,24 +4,30 @@ import android.app.Notification
 import android.app.NotificationChannel
 import android.app.NotificationManager
 import android.app.PendingIntent
-import android.app.Service
 import android.content.Context
 import android.content.Intent
 import android.os.Build
 import android.os.Handler
 import android.os.HandlerThread
-import android.os.IBinder
 import android.os.Looper
 import android.os.Message
 import android.os.Process
 import androidx.core.app.NotificationCompat
+import androidx.lifecycle.LifecycleService
+import androidx.lifecycle.Observer
+import com.twilio.video.Room.State.DISCONNECTED
 import com.twilio.video.app.ui.room.RoomActivity
+import com.twilio.video.app.ui.room.RoomEvent
+import com.twilio.video.app.ui.room.RoomEvent.RoomState
+import com.twilio.video.app.ui.room.RoomManager
+import dagger.android.AndroidInjection
 import timber.log.Timber
+import javax.inject.Inject
 
 const val VIDEO_SERVICE_CHANNEL = "VIDEO_SERVICE_CHANNEL"
 const val ONGOING_NOTIFICATION_ID = 1
 
-class VideoService : Service() {
+class VideoService : LifecycleService() {
```

Comment:
I am a little concerned about using the following dependencies in the service.

1. Dagger
2. Androidx Lifecycle

Dagger seems easy enough to refactor out, but it does seem a little trickier to remove the LifeCycle component based on this current implementation. Let's meet to discuss live.

Reply:
@aaalaniz Went ahead and took a crack at replacing LiveData with the Java standard library Observer pattern. We can review live tomorrow if you'd like.